### PR TITLE
fix: allow updating computed values of table widget in view mode

### LINKED ISSUE #9802: [Bug]: Table breaks with new data for a user
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

A user's app is broken

### Steps To Reproduce

1. Open the app https://app.appsmith.com/applications/61b9da60c70f2742893b91aa/pages/61b9da60c70f2742893b91ac?isCommentMode=false
2. Toggle the switch between the table data
3. The app does not show data even though the data is valid
4. This works in edit mode

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by sbalaji1192**
Fixes #9802
## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/table-computedvalues-view-mode 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.93 **(-0.01)** | 36.31 **(-0.01)** | 34.23 **(0)** | 55.46 **(0)**
 :red_circle: | app/client/src/utils/WorkerUtil.ts | 88.98 **(-0.78)** | 70.59 **(-1.96)** | 100 **(0)** | 92.38 **(-0.95)**
 :red_circle: | app/client/src/widgets/TableWidget/widget/index.tsx | 29.64 **(0)** | 9.48 **(-0.23)** | 29.41 **(0)** | 29.29 **(0)**</details>

### CODE REVIEW

File: app/client/src/pages/AppViewer/index.tsx [hunk 1]
```diff
@@ -150,6 +158,7 @@ class AppViewer extends Component<Props> {
               executeAction: this.props.executeAction,
               updateWidgetMetaProperty: this.props.updateWidgetMetaProperty,
               resetChildrenMetaProperty: this.props.resetChildrenMetaProperty,
+              batchUpdateWidgetProperty: this.props.batchUpdateWidgetProperty,
```
Comment (sbalaji1192) [inline app/client/src/pages/AppViewer/index.tsx:161]: Note: Exposing `batchUpdateWidgetProperty` in view mode as well.

File: app/client/src/widgets/TableWidget/widget/index.tsx [hunk 2]
```diff
@@ -582,7 +582,7 @@ class TableWidget extends BaseWidget<TableWidgetProps, WidgetState> {
 
     // If the user has changed the tableData OR
     // The binding has returned a new value
-    if (tableDataModified && this.props.renderMode === RenderModes.CANVAS) {
+    if (tableDataModified) {
```
Comment (sbalaji1192) [inline app/client/src/widgets/TableWidget/widget/index.tsx:585]: Note: now we're updating the computed values in the table widget regardless of rendermode. Enquire why we added this check/

Comment (SatishGandham) [reply app/client/src/widgets/TableWidget/widget/index.tsx:585]: I'm not why this was disabled in view mode. The computation itself doesn't look expensive, but, this update might trigger a render on the table.

Can you confirm if this piece code runs only when the tableData is actually modified

**Review by SatishGandham — APPROVED** [review]

File: app/client/src/widgets/TableWidget/widget/index.tsx (hunk 2 above)
Comment (Tooluloope) [reply app/client/src/widgets/TableWidget/widget/index.tsx:585]: Yeah I think it runs only when `santizedTableData` Changes

**Comment by YogeshJayaseelan** [comment]
@sbalaji1192 Verified this fix and it is working as expected

**Review by Tooluloope — APPROVED** [review]

**Comment by riodeuno** [comment]
I can't see a deploy preview here. Can we try the following test case?

- Create two buttons that adds two very different data to the `appsmith.store.testData` entity. 
- Bind this entity to the `tableData` 
- See what happens to the table columns, etc., when we click on each button?

There was a consideration due to which we wanted to not update the `primaryColumns` in view mode. I'm drawing a blank here regarding this, right now. Please test this thoroughly to make sure we're not regressing.

**Review by jsartisan — APPROVED** [review]

File: app/client/src/widgets/TableWidget/widget/index.tsx (hunk 2 above)
Comment (riodeuno) [reply app/client/src/widgets/TableWidget/widget/index.tsx:585]: This updates the primaryColumns, which we did not want in the view mode. @sbalaji1192 Now that we have the batchUpdateProperty available in view mode, please make sure that view mode does not change the original DSL. Basically, we need to make sure that these updates don't propagate back to the servers and modifies the DSL.

**Review by riodeuno — APPROVED** [review]
Please see my comment regarding the DSL updates. The rest looks good.

File: app/client/src/widgets/TableWidget/widget/index.tsx (hunk 2 above)
Comment (sbalaji1192) [reply app/client/src/widgets/TableWidget/widget/index.tsx:585]: @riodeuno Tested this on the DP. In view mode, we're not updating the DSL in the server. No save call is being made.

**Comment by YogeshJayaseelan** [comment]
@sbalaji1192 Tested this PR and new data does nto break table and also checked that there is no call to DSL

**Review by jsartisan — APPROVED** [review]

**Comment by somangshu** [comment]
@sbalaji1192 @jsartisan merged this overriding the flaky test fail as per the request

**Pull request merged**